extensions/stackdriver/stackdriver.cc

```diff
@@ -292,44 +316,76 @@ bool StackdriverRootContext::onDone() {
     done = false;
   }
   // TODO: add on done for edge.
+  for (auto const& item : tcp_request_queue_) {
+    // requestinfo is null, so continue.
+    if (item.second == nullptr) {
+      continue;
+    }
+    record(*item.second, true);
+  }
+  tcp_request_queue_.clear();
   return done;
 }
 
-void StackdriverRootContext::record() {
+bool StackdriverRootContext::record(::Wasm::Common::RequestInfo& request_info,
```

Comment:
Could you separate out TCP record logic to a different function (`recordTCP`)? And you can move the loop in onTick to that function.

Reply:
Sure..

Reply:
+1 for moving the logic of TCP to a different function..
but moving the loop in onTick to that function might be tricky.. we call record at the end of a connection too, to make sure this connection does gets recorded if ticker didn't pick it up.. 
will add a different function for loop that onTick can use..so that code their is concise..

@mandarjog : thinking about this, I remember you mentioned that we always want to capture connection open.. currently, we call record on ticker and when connection ends. Is that enough? I feel if connection is going to last for long, ticker will most likely pick it up and then connection end record is always there...

Reply:
Ok I see. Yeah then keeping the loop out of the record call makes sense to me.

Record on open would introduce some overhead, so better avoid doing that.

Reply:
We need to record at begin , ticker and end  for tcp. When keep alives are used we should see one tcp connection and many http on top of it.

Reply:
We already record at beginning. We only record connection open event at new connection now (probably will need to record the time as well, but that could happen later), instead of the full connection information. Those information could be backfilled ontick or onclose, which is the current behavior and I think it is good enough.